### PR TITLE
Polished up comments around HasSynced()

### PR DESCRIPTION
**Pull request by jqmichael**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind documentation

**What this PR does / why we need it**:
Polished up comment around `HasSynced()`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by jqmichael** [comment]
/assign @caesarxuchao

@MikeSpreitzer

File: staging/src/k8s.io/client-go/tools/cache/fifo.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (f *FIFO) Close() {
 }
 
 // HasSynced returns true if an Add/Update/Delete/AddIfNotPresent are called first,
-// or an Update called first but the first batch of items inserted by Replace() has been popped
+// or the first batch of items inserted by Replace() has been popped.
```
Comment (jqmichael) [inline staging/src/k8s.io/client-go/tools/cache/fifo.go:148]: As long as the first `Replace()` is called first, it doesn't really matter whether `Add/Update/Delete` are called after. All it maters is whether the first batch of items are popped. Therefore I made the comments shorter and easier to follow.

**Review by MikeSpreitzer — COMMENTED** [review]
The changes here are improvements.
The comment on FIFO::populated could also be updated.

**Pull request merged**